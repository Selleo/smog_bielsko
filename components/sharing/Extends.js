import { Dimensions } from 'react-native'

import Env            from '../../env'

export default stationIdx = 6535;

export function fetchBgr(query) {
  url = 'https://www.googleapis.com/customsearch/v1?key=' + Env.GOOGLE_KEY + '&cx=' + Env.GOOGLE_CX + '&q=' + encodeURI(query) + '&imgType=photo&searchType=image&fileType=jpg&imgSize=large&num=1&alt=json';
  return fetch(url)
    .then((response) => response.json())
    .then((response) => response.items[0])
}

export function getData() {
  return fetch('http://api.waqi.info/feed/@' + stationIdx + '/?token=' + Env.API_KEY)
    .then((response) => response.json())
}
