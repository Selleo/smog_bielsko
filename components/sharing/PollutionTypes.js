export const colors = {
  1: '#57b108',
  2: '#b0dd10',
  3: '#ffd911',
  4: '#e58100',
  5: '#d82329'
};

export const names = {
  'co': 'Tlenek Węgla',
  'h': 'Wilgotność',
  'no2': 'Dwutlenek Azotu',
  'o3': 'Ozon',
  'p': 'Ciśnienie',
  'pm10': 'Pył zawieszony (PM10)',
  'so2': 'Dwutlenek Siarki',
  't': 'Temperatura',
  'w': 'Wiatr'
};

export const icons = {
  't': "thermometer",
  'p': "pressure",
  'h': "rain",
  'w': "wind",
};

export const suffices = {
  't': '°C',
  'p': 'hPa',
  'h': '%',
  'w': ' m/s'
};
