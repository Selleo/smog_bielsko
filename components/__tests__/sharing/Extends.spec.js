import 'react-native'
import React                            from 'react'

import chai, { expect }                 from 'chai'
import dirtyChai                        from 'dirty-chai'
import fetchMock                        from 'fetch-mock'

import Env                              from '../../../env'
import { stationId, fetchBgr, getData } from '../../sharing/Extends'
import { data, image, imageResponse }   from './../mock/data'

const originalExpect = global.expect;
chai.use(dirtyChai);

describe('Index', () => {
  it('renders correctly', () => {
    originalExpect(stationId).toMatchSnapshot();
    expect(stationId).to.exist();
    originalExpect(fetchBgr).toMatchSnapshot();
    expect(fetchBgr).to.exist();
    originalExpect(getData).toMatchSnapshot();
    expect(getData).to.exist();
  });

  describe('stationId', () => {
    it('is number', () => {
      expect(stationId).to.equal(6535);
      expect(stationId).to.to.be.a('number');
    });
  });

  describe('#fetchBgr', () => {
    it('is function', () => {
      expect(fetchBgr).to.to.be.a('function');
    });

    it('fetch image from google API', (done) => {
      let url = 'https://www.googleapis.com/customsearch/v1?key=' + Env.GOOGLE_KEY + '&cx=' + Env.GOOGLE_CX + '&q=Zywiec%20city&imgType=photo&searchType=image&fileType=jpg&imgSize=large&num=1&alt=json';
      let response = undefined;

      fetchMock.get(url, { items: image });

      let request = fetchBgr('Zywiec city');

      setTimeout(() => {
        request.then((item) => {
          response = item;
        });
      }, 1);

      setTimeout(() => {
        expect(response.htmlSnippet).to.equal("Old <b>Town</b> - <b>Bielsko</b>-<b>Biala</b>");
        done();
      }, 2);

      fetchMock.restore();
    });
  });

  describe('#getData', () => {
    it('is function', () => {
      expect(getData).to.to.be.a('function');
    });

    it('fetch station data', (done) => {
      let url = 'http://api.waqi.info/feed/@6535/?token=' + Env.API_KEY;
      let response = undefined;

      fetchMock.get(url, { data: data });

      let request = getData(6535);

      setTimeout(() => {
        request.then((item) => {
          response = item;
        });
      }, 1);

      setTimeout(() => {
        expect(response.data.city.name).to.equal('Bielsko-Bia≈Ça, ul. Kossak-Szczuckiej 19');
        done();
      }, 2);

      fetchMock.restore();
    });
  });

});
