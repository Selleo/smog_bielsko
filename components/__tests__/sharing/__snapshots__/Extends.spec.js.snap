exports[`Index renders correctly 1`] = `6535`;

exports[`Index renders correctly 2`] = `[Function]`;

exports[`Index renders correctly 3`] = `[Function]`;
